src/lib/mina_net2/libp2p_stream.ml

```diff
@@ -126,6 +126,44 @@ let stream_closed ~logger ~who_closed t =
         ] ;
   `Stream_should_be_released (equal_state FullyOpen t.state)
 
+let max_message_size = 16777216 (* 16 MiB *)
+
+let split_string ~every b =
+  let blen = String.length b in
+  let rec isplit sofar pos =
+    if blen - pos > every then
+      let chunk = String.sub b ~pos ~len:every in
+      isplit (chunk :: sofar) @@ (pos + every)
+    else if blen - pos = 0 then sofar
+    else String.sub ~pos ~len:(blen - pos) b :: sofar
+  in
+  List.rev (isplit [] 0)
+
+let%test_unit "split_string" =
+  let gen =
+    let module Gen = Quickcheck.Generator in
+    let%bind.Gen every_ = Gen.small_positive_int in
+    let%bind.Gen total_ = Gen.small_non_negative_int in
+    let every, total =
+      if total_ < every_ then (every_, total_) else (total_, every_)
```

Comment:
`every_ > 0` by definition and `total_` might be zero.

In case `total_ = 0`,  `total_ < every_` and `every := every_ > 0` and `total := total_ = 0`.

Hence `every = 0` is not possible in this case. It's not possible in other cases as well.

Reply:
Okay, maybe it doesn't make sense :joy: 

Like, yeah, condition `total_ < every_` was meant to be the opposite (otherwise test is kind of dumb), but then the case you mention would be possible.

I\ll update the code of the test.

Reply:
I will just remove this logic of checking that `every < total`.

It's fine to test function in both cases `every > total` and `every <= total` as long as `every > 0`